sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveInlineTables.scala

```diff
@@ -95,10 +95,14 @@ object ResolveInlineTables extends Rule[LogicalPlan] {
       InternalRow.fromSeq(row.zipWithIndex.map { case (e, ci) =>
         val targetType = fields(ci).dataType
         try {
-          if (e.dataType.sameType(targetType)) {
-            e.eval()
+          val castedExpr = if (e.dataType.sameType(targetType)) {
+            e
           } else {
-            Cast(e, targetType).eval()
+            Cast(e, targetType)
+          }
+          castedExpr match {
+            case te: TimeZoneAwareExpression => te.withTimeZone(conf.sessionLocalTimeZone).eval()
```

Comment:
Or did `ResolveTimeZone` already process the input expression? In that case just add the timezone to the cast.